build/xtask/src/config.rs

```diff
@@ -348,6 +348,58 @@ impl Config {
             );
         }
 
+        if let Some(ext_int_ctrl_toml) = self.tasks.get("ext_int_ctrl") {
+            let ext_interrupts = ext_int_ctrl_toml
+                .config
+                .as_ref()
+                .expect("External Interrupt Controller config is missing")
+                .as_table()
+                .expect("External Interrupt Controller config is not a struct");
+
+            let val_into_array = |name: &str| -> &Vec<ordered_toml::Value> {
+                ext_interrupts
+                    .get(name)
+                    .expect(format!("External Interrupt Controller config missing field: {}", name).as_str())
+                    .as_array()
+                    .expect(format!("External Interrupt Controller config field {} is not an array", name).as_str())
+            };
+
+            let notifications = val_into_array("notification");
+            let task_names = val_into_array("tasks");
+            let source_names = val_into_array("source");
+
+            for (i, name_val) in task_names.iter().enumerate() {
+                let name = name_val
+                           .as_str()
+                           .expect(format!("External Interrupt Controller config field \"name\" entry {} was not a string", i).as_str())
+                           .strip_prefix("\"")
+                           .expect(format!("External Interrupt Controller config field \"name\" entry {} missing leading quotation mark", i).as_str())
+                           .strip_suffix("\"")
+                           .expect(format!("External Interrupt Controller config field \"name\" entry {} missing closing quotation mark", i).as_str());
+                let source = source_names[i]
+                             .as_str()
+                             .expect(format!("External Interrupt Controller config field \"source\" entry {} was not a string", i).as_str())
+                             .strip_prefix("\"")
+                             .expect(format!("External Interrupt Controller config field \"source\" entry {} missing leading quotation mark", i).as_str())
+                             .strip_suffix("\"")
+                             .expect(format!("External Interrupt Controller config field \"source\" entry {} missing closing quotation mark", i).as_str());
+                let notif = notifications[i]
+                            .as_integer()
+                            .expect(format!("External Interrupt Controller config field \"notification\" entry {} was not an integer", i).as_str());
+
+                if name == task_name {
+                    notif_consts.push_str("const ");
+                    notif_consts.push_str(
+                        format!("{}_", source.to_ascii_uppercase()).as_str(),
+                    );
+                    notif_consts.push_str(
+                        format!("NOTIFICATION: u32 = 0x{:X};\n", notif)
```

Comment:
`IRQ` is technically the name of that irq, as defined in the [chip.toml](https://github.com/rivosinc/hubris/blob/ef3d9479e4cb80ba1fe145fa008f313246eea385/chips/fe310-g003/chip.toml#L20). This must be used as it's possible for a device to provide multiple interrupt sources. I can add `IRQ` for external interrupts as well, but if there are going to be peripherals sending multiple interrupts the naming won't be consistent.

Reply:
I was thinking from a usage standpoint. There would be IRQ because of `plic.irq` but no IRQ in the name because of `rtc`. Anyways, we can keep this as is for now, but if it starts getting confusing we should probably consider adding the _IRQ_ in name or adopt a similar naming for external interrupts with `rtc.irq` in app.toml